tests/update_state_test.py

```diff
@@ -0,0 +1,202 @@
+        assert game.code_number == 0
+
+
+class TestCheckWin:
+    '''Check_win is responsible for assigning a winner if the death card 
+    is picked or if the entire deck has been flipped'''
+    def test_winner_and_game_over_for_death_card(self):
+        'test if winner_color is the opposite team and game_over is True'
+        game = Game()
+        game.guess_value = 4
+        game.team = "blue"
+
+        game.check_win()
+
+        assert game.winner_color == 'red'
+        assert game.game_over is True
+
+    def test_winner_for_death_card(self):
+        'test if when team is "red", winner_color is "blue"'
+        game = Game()
+        game.guess_value = 4
+        game.team = "red"
+
+        game.check_win()
+
+        assert game.winner_color == 'blue'
+        assert game.game_over is True
+
+    def test_red_complete_deck(self):
+        'test if when all red cards are flipped, winner_color = red and game_over is True'
+        game = Game()
+        # make all red cards guessed
+        for card in game.board.cards:
+            if card.color == 'red':
+                card.flipped = True
+
+        game.check_win()
+
+        assert game.winner_color == 'red'
+        assert game.game_over is True
+
+    def test_blue_complete_deck(self):
+        'test if when all blue cards are flipped, winner_color = blue and game_over is True'
+        game = Game()
+        # make all blue cards guessed
+        for card in game.board.cards:
+            if card.color == 'blue':
+                card.flipped = True
+
+        game.check_win()
+
+        assert game.winner_color == 'blue'
+        assert game.game_over is True
+
+class TestUpdateState:
+    '''Update_state sets guessing to True or False, then sets skip True first then False,
+    and finally, switches the team color and resets guess_value to zero'''
+    def test_guessing_is_true_under_right_conditions(self):
+        'If these three variables are as they are set here, guessing should be True'
+        game = Game()
+        game.guess_value = 1
+        game.code_number = 1
+        game.game_over = False
+
+        game.update_state()
+
+        assert game.guessing is True
+
+    def test_game_over_and_skip_are_true_makes_guessing_false(self):
+        'When game_over is True guessing is False. When skip is True, it is changed to False'
+        'Team color is changed from blue to red. Guess_value is reset'
+        game = Game()
+        game.guess_value = 1
+        game.code_number = 1
+        game.game_over = True
+        # need skip to be True so that self.guessing is not reset to True before test
+        game.skip = True
+        game.team = "blue"
+
+        game.update_state()
+
+        assert game.skip is False
+        assert game.guessing is False
+        assert game.team == "red" # team color should be switched if game.skip is False
+        assert game.guess_value == 0 # guess_value should reset if game.skip is False
```

Comment:
If the game is over, do you really care what all these values are? You should probably only care about whatever is needed to drive one last proper `game.render()` call. Test that your code works the way it needs to work, not that it works the way it does work. Leave your solution as unconstrained by tests as possible while still proving it works. That way you have flexibility to change your solution without breaking your tests. This will be important as your port from Python to C, as you may have to pick new strategies in C.

Reply:
That makes sense. I changed it to make sure `game.skip` and  `game.guessing` are true which is what is necessary for `game.render( )` to correctly makes its last render.